File: solutions/scala/gigasecond/1/src/main/scala/Gigasecond.scala
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.Instant
import java.time.ZoneId

object Gigasecond {
	private val gigaSecondInMillis = 1000000000L * 1000
	private val zoneId = ZoneId.systemDefault()

  def add(startDate: LocalDate): LocalDateTime = {
  	val epoch = startDate.atStartOfDay(zoneId).toEpochSecond() * 1000L
  	LocalDateTime.ofInstant(Instant.ofEpochMilli(epoch + gigaSecondInMillis + 3600 * 1000), zoneId)
  }

  def add(startDateTime: LocalDateTime): LocalDateTime = {
  	val epoch = startDateTime.atZone(zoneId).toEpochSecond() * 1000L
  	LocalDateTime.ofInstant(Instant.ofEpochMilli(epoch + gigaSecondInMillis - 3600 * 1000), zoneId)
  }
}

File: solutions/java/gigasecond/1/src/main/java/Gigasecond.java
import java.math.BigInteger;
import java.time.LocalDate;
import java.time.LocalDateTime;

public class Gigasecond {
    private LocalDateTime dtm;
    private static BigInteger _GIGASECOND = BigInteger.valueOf(1_000_000_000L);

    public Gigasecond(LocalDate moment) {
        dtm = moment.atStartOfDay();
    }

    public Gigasecond(LocalDateTime moment) {
        dtm = moment;
    }

    public LocalDateTime getDateTime() {
       return dtm.plusSeconds(_GIGASECOND.longValue());
    }
}
